src/Common/src/System/Globalization/FormatProvider.Number.cs

```diff
@@ -421,7 +421,7 @@ private static unsafe bool ParseNumber(ref char* str, char* strEnd, NumberStyles
                             {
                                 if (bigNumber)
                                 {
-                                    sb.Append(ch);
+                                    sb!.Append(ch);
```

Comment:
It can't infer this from `bool bigNumber = (sb != null); ` ? Is there a way to educate the compiler on that line, like `[NotNullIfTrue]` or something?

Reply:
So that to review `sb!` lower down I don't have to scroll up.

Reply:
I've raised this question before and they said that unfortunately in this cases the compiler can't track nullability from a cached result.